### ISSUE TITLE
False Positive: Style/CollectionCompact offense on Set#reject!

### ISSUE THREAD

**Issue by pdobb** [rubocop/rubocop#13738]
Ruby's [Set](https://rubyapi.org/3.4/o/set) class doesn't define `compact!`. 
Assuming we want to alter the original Set instance, the best we can do is `reject!(&:nil?)`.

But Style/CollectionCompact reports an offense:

```ruby
C: [Correctable] Style/CollectionCompact: Use compact! instead of reject!(&:nil?).
    Set.new.reject!(&:nil?)
            ^^^^^^^^^^^^^^^
```

--------

## Expected behavior

Don't report an offense when calling `.reject(&:nil?)` on a Set object.

## Actual behavior

```ruby
C: [Correctable] Style/CollectionCompact: Use compact! instead of reject!(&:nil?).
    Set.new.reject!(&:nil?)
            ^^^^^^^^^^^^^^^
```

## Steps to reproduce the problem

```ruby
Set.new.reject!(&:nil?)

# OR

set = Set.new
set << nil
set.reject!(&:nil?)

# OR

["whatever"].inject(Set.new) { |acc, element| acc << nil }.reject!(&:nil?)
```

## RuboCop version

```
$ bundle exec rubocop -V
1.70.0 (using Parser 3.3.7.0, rubocop-ast 1.37.0, analyzing as Ruby 3.4, running on ruby 3.4.1) [arm64-darwin24]
  - rubocop-minitest 0.36.0
  - rubocop-performance 1.23.1
  - rubocop-rails 2.29.0
  - rubocop-rake 0.6.0
```
